doc/sandbox/Losses-restofit.rst

```diff
@@ -0,0 +1,391 @@
+    >>> solver = rr.FISTA(problem)
+    >>> vals = solver.fit()
+    >>> solver.composite.coefs
+
+
+::
+
+
+    ---------------------------------------------------------------------------
+
+    AttributeError                            Traceback (most recent call last)
+
+    <ipython-input-64-82e5a9292d8d> in <module>()
+    ----> 1 problem = rr.simple_problem(quadratic, hinge_loss)
+          2 solver = rr.FISTA(problem)
+          3 vals = solver.fit()
+          4 solver.composite.coefs
+
+
+    /home/jb/code/regreg/regreg/problems/simple.py in __init__(self, smooth_atom, proximal_atom)
+         20         self.smooth_atom = smooth_atom
+         21         self.proximal_atom = proximal_atom
+    ---> 22         self.coefs = self.smooth_atom.coefs = self.proximal_atom.coefs
+         23 
+         24     def smooth_objective(self, x, mode='both', check_feasibility=False):
+
+
+    AttributeError: 'affine_atom' object has no attribute 'coefs'
+
+
+.. nbplot::
+
+    >>> quadratic.coefs
+    array([ 0.,  0.,  0.])
+
+This determines a line in the plane, specified as
+:math:`\beta_1 \cdot x + \beta_2 \cdot y + \gamma = 0` and the
+classifications are determined by which side of the line a point is on.
+
+.. nbplot::
+
+    >>> fits = np.dot(X_1, problem.coefs)
+    >>> labels = 2 * (fits > 0) - 1
+    >>> accuracy = (1 - np.fabs(Y-labels).sum() / (2. * N))
+    >>> accuracy
+
+.. nbplot::
+
+    >>> import numpy as np
+    >>> import regreg.api as rr
+    >>>
+    >>> np.random.seed(400)
+    >>>
+    >>> N = 500
+    >>> P = 2
+    >>>
+    >>> Y = 2 * np.random.binomial(1, 0.5, size=(N,)) - 1.
+    >>> X = np.random.standard_normal((N,P))
+    >>> X[Y==1] += np.array([3,-2])[np.newaxis,:]
+    >>>
+    >>> X_1 = np.hstack([X, np.ones((N,1))])
+    >>> X_1_signs = -Y[:,np.newaxis] * X_1
+    >>> transform = rr.affine_transform(X_1_signs, np.ones(N))
+    >>> C = 0.2
+    >>> hinge = rr.positive_part(N, lagrange=C)
+    >>> hinge_loss = rr.linear_atom(hinge, transform)
+    >>>
+    >>> quadratic = rr.quadratic.linear(rr.selector(slice(0,P), (P+1,)), coef=0.5)
+    >>> problem = rr.simple_problem(quadratic, hinge_loss)
+    >>> solver = rr.FISTA(problem)
+    >>> solver.fit()
+    >>>
+    >>> import pylab
+    >>> pylab.clf()
+    >>> pylab.scatter(X[Y==1,0],X[Y==1,1], facecolor='red')
+    >>> pylab.scatter(X[Y==-1,0],X[Y==-1,1], facecolor='blue')
+    >>>
+    >>> fits = np.dot(X_1, problem.coefs)
+    >>> labels = 2 * (fits > 0) - 1
+    >>>
+    >>> pointX = [X[:,0].min(), X[:,0].max()]
+    >>> pointY = [-(pointX[0]*problem.coefs[0]+problem.coefs[2])/problem.coefs[1],
+    ...           -(pointX[1]*problem.coefs[0]+problem.coefs[2])/problem.coefs[1]]
+    >>> pylab.plot(pointX, pointY, linestyle='--', label='Separating hyperplane')
+    >>> pylab.title("Accuracy = %0.1f %%" % (100-100 * np.fabs(labels - Y).sum() / (2 * N)))
+    >>> #pylab.show()
+
+Sparse SVM :sub:`:sub:`~`\ ~`\ ~~
```

Comment:
This should have been an underline under `Sparse SVM` but some converter got confused.